actionpack/lib/action_dispatch/middleware/host_authorization.rb

```diff
@@ -56,17 +60,45 @@ def sanitize_string(host)
         end
     end
 
-    DEFAULT_RESPONSE_APP = -> env do
-      request = Request.new(env)
+    class DefaultResponseApp # :nodoc:
+      RESPONSE_STATUS = 403
+
+      def call(env)
+        request = Request.new(env)
+        format = request.xhr? ? "text/plain" : "text/html"
+
+        if request.local? || request.get_header("action_dispatch.show_detailed_exceptions")
+          response(format, template_body(request))
+        else
+          log_error(request)
+          response(format)
+        end
```

Comment:
Looking at two related middlewares:
https://github.com/rails/rails/blob/25f6a8e84fb6c27f489fc3e593656ddb72f25613/actionpack/lib/action_dispatch/middleware/actionable_exceptions.rb#L26
https://github.com/rails/rails/blob/25f6a8e84fb6c27f489fc3e593656ddb72f25613/actionpack/lib/action_dispatch/middleware/debug_exceptions.rb#L59-L73

I think we can do the following:

1. Log the error in both cases.
2. Only look at the "action_dispatch.show_detailed_exceptions" header (if it's good enough for actionable errors...).

```suggestion
        log_error(request)
          
        if request.get_header("action_dispatch.show_detailed_exceptions")
          response(format, template_body(request))
        else
          response(format)
        end
```

Reply:
Thanks for the suggestions. Agree on both